terragrunt/aws/cloudfront/cloudfront.tf

```diff
@@ -0,0 +1,55 @@
+resource "aws_cloudfront_origin_access_identity" "origin_access_identity" {
+  comment = "cloudfront origin access identity"
+}
+
+resource "aws_cloudfront_distribution" "distribution" {
+  for_each            = var.s3_bucket_regional_domain_name
+  default_root_object = "index.html"
+  enabled             = true
+  web_acl_id          = aws_wafv2_web_acl.cds_website_waf.arn
+
+  default_cache_behavior {
+    target_origin_id       = each.value
+    viewer_protocol_policy = "redirect-to-https"
+    allowed_methods        = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods = [
+      "GET",
+      "HEAD",
+    ]
+    forwarded_values {
+      query_string = false
+
+      cookies {
+        forward = "all"
+      }
+    }
+  }
+  origin {
+    domain_name = each.value
+    origin_id   = each.value
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.origin_access_identity.cloudfront_access_identity_path
+    }
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+
+  }
+
+  viewer_certificate {
```

Comment:
We'll need a viewer certificate here so that people can use the custom domains to access the CloudFront dist:
https://github.com/cds-snc/gcds-terraform/blob/1ffb8d00f6b34da5d9d8eec9f262d1abbe96f56b/terragrunt/aws/cdn/cloudfront.tf#L39

Here's what the ACM definition looks like:
https://github.com/cds-snc/gcds-terraform/blob/1ffb8d00f6b34da5d9d8eec9f262d1abbe96f56b/terragrunt/aws/cdn/acm.tf